acmens.py

```diff
@@ -145,17 +145,28 @@ def _do_challenge(challenge_type, authz_url, nonce_url, auth, account_key, thumb
     # Choose challenge.
     preferred_type = "dns-01" if challenge_type == "dns" else "http-01"
     challenge = None
+    dns_challenge = None
     http_challenge = None
     for c in chl_result["challenges"]:
         if c["type"] == preferred_type:
             challenge = c
+        if c["type"] == "dns-01":
+            dns_challenge = c
         if c["type"] == "http-01":
             http_challenge = c
     if challenge is None:
         if http_challenge is None:
-            sys.stderr.write("Error: Unable to find challenges!")
```

Comment:
Can we refactor it like this:

```python
if http_challenge:
    # ....
else:
     if dns_challenge:
           # ....
     else:
           # ....
```

Reply:
So, from this
```py
    if challenge is None:
        if http_challenge is None:
            if dns_challenge is None:
                sys.stderr.write("Error: Unable to find challenges!")
                sys.exit(1)
            else:
                # Fallback to dns challenge.
                challenge = dns_challenge
                challenge_type = "dns"
        else:
            # Fallback to http challenge.
            challenge = http_challenge
            challenge_type = "http"
```

To this?
```py
    if challenge is None:
        if http_challenge:
            # Fallback to http challenge.
            challenge = http_challenge
            challenge_type = "http"
        else
            if dns_challenge:
                # Fallback to dns challenge.
                challenge = dns_challenge
                challenge_type = "dns"
            else:
                sys.stderr.write("Error: Unable to find challenges!")
                sys.exit(1)
```

Yes, that should be equivalent. I just tried to keep the initial order/construct as much as possible. But while we're at it, I'd suggest the following to remove one level:
```py
    if challenge is None:
        if http_challenge:
            # Fallback to http challenge.
            challenge = http_challenge
            challenge_type = "http"
        elif dns_challenge:
            # Fallback to dns challenge.
            challenge = dns_challenge
            challenge_type = "dns"
        else:
            sys.stderr.write("Error: Unable to find challenges!")
            sys.exit(1)
```